src/org/apache/html/dom/HTMLDOMImplementationImpl.java

```diff
@@ -82,7 +82,7 @@ public final HTMLDocument createHTMLDocument( String title )
      * used to perform operations that are not specific to a particular
      * document instance, e.g. to create a new document.
      *
-     * @return Reference to a valid DOM implementation
+     * @return a singleton reference to a valid DOM implementation
```

Comment:
how do you know this is a singleton?

Reply:
because *_instance* is static final *HTMLDOMImplementationImpl* there'll only be one. I can adjust the wording but this is the singleton pattern of a constant holding an instance which is returned via a static method. The constructor is private so this is the only class where `new HTMLDOMImplementationImpl()` can be called.